### ISSUE TITLE
Can't build

### ISSUE THREAD

**Issue by r14n** [RSully/mac-cec-volume#4]
Hi @RSully , admittedly I'm new to Xcode (I'm a web developer).  I seem to be missing some things from the project, or perhaps something is out of date, or maybe something is too current.  I could use some help.  My goal is just to control my TV (I use it as a monitor) volume using the keyboard, I can't find your software to download anywhere but here on github.

I'm getting build errors, the first is on CECAPI.mm line 10 #import <cec.h>

I'm thinking I need to get the older version of libcec but I don't know where to find it.

I'd be happy enough with a prebuilt version of your software if it will work with my setup.

MacBook Air (13-inch, Early 2015)
Intel HD Graphics 6000 1536 MB
macOS 10.13.3 (17D47)
Xcode Version 9.3 (9E145)

libCEC version: 4.0.2, git revision: <unknown>, compiled on Mon Sep 18 23:38:30 UTC 2017 by brew@HighSierra.local on Darwin 17.0.0 (x86_64), features: P8_USB, DRM

I'll mention (although I'm not up to that step yet) I am using the Pulse8 adapter and I have success controlling the computer with the TV remote via the app Couch Slouch.

**Comment by RSully** [comment]
I'll try to help the best I can. I haven't touched this project in a while sadly, and I'm not going to be providing pre-built versions anytime soon. It only exists up on GitHub for now.

When I last worked on this project libCEC was at version 2.x (with 3 just being released). It looks like they're up to version 4 now, so I'll need to update this project if it has changed enough. That could be a problem. For now, you could try installing version 2.2.0. One way to do that would be to uninstall libcec, download the old version of the formula rb file, put it in your homebrew formulas folder, and then issue the install command. I did a quick search and found [this formula](https://github.com/Homebrew/legacy-homebrew/blob/4a795cbdf0cc03cc2179fe78d6be340b9ad10eda/Library/Formula/libcec.rb). Not sure if it'll work but it's worth a shot. 

Additionally, you'll need to make sure the header and library include paths are correct. I have homebrew in a weird location (e.g. not /usr/local), and that is saved in the Xcode project file. I'm not sure if this matters, /usr/local *may* be a default search path, I'm not 100% certain. 

I didn't know libcec was up to v4 so if the above doesn't work or is too much work, I'll try to give this project some attention in the next month or so. I was getting frustrated with the library and lost interest, but maybe it's better now.

**Comment by r14n** [comment]
Thank you.

I managed to get all that done.  The paths were a bit tricky to get sorted but it seems I've got it right.

However it doesn't seem to actually be controlling the volume on the TV, not sure why.  

2018-04-05 22:57:27.710816-0400 HDMI Volume Control[926:10629] found /dev/cu.usbmodemv1
2018-04-05 22:57:27.710949-0400 HDMI Volume Control[926:10629] Opening /dev/cu.usbmodemv1
2018-04-05 22:57:30.886333-0400 HDMI Volume Control[926:10629] Finished setting up CEC (0)
2018-04-05 22:57:37.452213-0400 HDMI Volume Control[926:10629] Sound up
2018-04-05 22:57:38.752209-0400 HDMI Volume Control[926:10629] Sound down
2018-04-05 22:57:39.306761-0400 HDMI Volume Control[926:10629] Sound up
2018-04-05 22:57:39.807701-0400 HDMI Volume Control[926:10629] Sound up

Pressing the Debug button yields:

2018-04-05 23:02:36.529150-0400 HDMI Volume Control[961:12372] primary device: 15


Any suggestions?

**Comment by RSully** [comment]
Make sure your adapter is plugged in the correct direction on your HDMI cables. Remember you want to control the TV from your laptop. It could be sending the commands to your Mac if its in the wrong direction, and the Mac will ignore them. Additionally, make sure your TV has CEC enabled and all that fun stuff (and that it actually supports being controlled, and not just being the controlling one).

**Comment by r14n** [comment]
Good call.  I guess it was originally pointing in the "wrong" direction as I plugged it in as marked on the adapter.  Their intention is for the TV remote to control the computer, and we want the opposite.

Reversing it didn't work however.  It's possible my TV doesn't support it - I can't really test that right now as I don't have another device which I know to send CEC commands.  (VIZIO D50u-D1) All I can find on the web is that CEC is "Yes".

I did also verify CEC is enabled on the TV, although I already knew it was since I am able to control the Mac with the TV remote through Couch Slouch when the cable was plugged in the "wrong" way.

To attempt a different setup to test this I put the Pulse8 between the HDMI ARC Out on the TV and the HDMI ARC on the Soundbar (also Vizio).  The Mac is still outputting video and audio to the TV.  The TV is still able to send the audio and CEC volume commands to the soundbar (passed through the Pulse8) but I'm not able to get the Mac to control the soundbar volume in this way.  Also tried reversing the Pulse8 input output HDMI cables in this setup.

Edit: should we be concerned that  Finished setting up CEC (0) reports 0?  Or is that the success code?

**Comment by RSully** [comment]
Yeah good call, 0 is a fail I believe. Make sure nothing else is open that is using the adapter. If you have a media application open that supports CEC try closing it. It is probably the case that only 1 application can have the device open at the time (perhaps a newer version of libCEC fixes this, I'm not sure).

**Comment by RSully** [comment]
I just pushed up a wip-libcec-v4 branch to GitHub. You can try that with libcec v4. It compiles fine, but I have not tested it yet. Don't uninstall v2 though (or do a brew clean or anything that would delete it) so that you can continue using the old version too.

**Comment by r14n** [comment]
Thanks, I won't get to try it until Thursday, but I'm looking forward t
solving this puzzle.

**Comment by r14n** [comment]
It's working!  I'll keep you posted with any new info.

**Comment by RSully** [comment]
Did using the new branch make it work, or something else? Definitely interested in what changed!

Edit: and did you test the new branch?

**Comment by r14n** [comment]
Yes, the first run of the new branch worked.

**Comment by RSully** [comment]
If you could, try the old build as well just to see if something external changed or not.

**Comment by r14n** [comment]
I did brew switch 2.2.0, still having the problem with it not finding the device.

2018-04-17 08:34:27.789352-0400 HDMI Volume Control[31489:1114342] Finished setting up CEC (0)

Thank you so much for your v4 branch, I'm absolutely loving having control of my volume from the keyboard!
